packages/zipkin-transport-http/package.json

```diff
@@ -14,12 +14,13 @@
   "license": "Apache-2.0",
   "repository": "https://github.com/openzipkin/zipkin-js",
   "dependencies": {
+    "node-fetch": "^2.6.0",
```

Comment:
I am kind of concerned about this version constraint. I checked node-fetch api and nothing stop us to use [version 1.4.0](https://github.com/node-fetch/node-fetch/tree/v1.4.0) for example since it support all the options we use. In the other hand, [version 3.x is going to break us](https://github.com/node-fetch/node-fetch/commit/94e5b92de1d7bb3506ac6487581963a327e00d22#diff-2936d6bde71136a082ca21cd43f997e9R26). I feel like we could support ^1.4 and ^2.0? What do you think. I don't want us to force users to upgrade their node-fetch because of zipkin.

Reply:
why do you think this package uses 1.4.0 ? I moved this package from version 2.3 to 2.6 
you suggest to fix it here as ^2.0.0?

Reply:
> why do you think this package uses 1.4.0 ? I moved this package from version 2.3 to 2.6
> you suggest to fix it here as ^2.0.0?

I think you could specify it as `>=2.0.0 < 3.0.0` (could also be that simply `2.x` expresses the same thing but don't quote me on that, I've slipped up with semver constraints before)

Could also be `>=1.4.0 < 3.0.0` to include the 1.4.0 version as @jcchavezs mentioned

Reply:
Did I get it right that ^2.0.0 will be fine for everyone?

Reply:
> Did I get it right that ^2.0.0 will be fine for everyone?

Would be fine by me personally because I don't use old node-fetch (`v1.4.x`) at all.

Reply:
Yeah, my main concern are those using 1.4.x that we will upgrade them to 2.0.x but now I think that since this is not a breaking change because we are just adding the dependency it is fine to keep ^2.0.0. Any thoughts @adriancole?

Reply:
> Yeah, my main concern are those using 1.4.x already that want to use zipkin-js, will that force them to use node-fetch 2.x then? 
>In the other hand I think that since this is not a breaking change because we are just adding the dependency it is fine to keep ^2.0.0. Any thoughts @adriancole?

If they want to upgrade to the newer version of zipkin-js, then yes, they will be forced to update node-fetch from 1.4.x to at least 2.x.
So, in this sense, this is a breaking change and should be accompanied by a major version bump on the version of zipkin-js itself.

Basically it's like this: If you previously did not restrict a peer dependency version, but now you do, then it is a breaking change for those who are suddenly out of compliance with their dependency version.

If it were released as a minor/patch version bump then it might break builds for some, if they had minor/patch version auto upgrade specified in their zipkin-js version in the package.json dependencies section. This is how builds of projects with npm dependency management can break overnight seemingly all by themselves without anyone changing anything in the source code.

So, I'd say it needs a major release.
That way people who are upgrading their zipkin-js will know to "expect trouble" and those who specified to only auto-update patch/minor versions won't get a sudden and unexpected breakage.

People who specified to auto-upgrade even major versions and use node-fetch 1.4.x will still have their build break overnight, but they chose to live on the bleeding edge so there's nothing you can do about that.

Reply:
Thanks for the explanation @petermetz, if that is the case then let's do `>=1.4.0 < 3.0.0` so we don't break anyone. We are not likely doing a major release for this change.

Reply:
3448533b8a01daf27da88f7e1b3e9fc9e8a7b62d